comms/src/builder/builder.rs

```diff
@@ -0,0 +1,491 @@
+        let config = self.omp_config_factory.take().map(|f| f.make()).unwrap_or_default();
+
+        OutboundMessagePool::new(
+            config,
+            self.comms_context.clone(),
+            // OMP can requeue back onto itself
+            message_sink_address.clone(),
+            message_sink_address.clone(),
+            peer_manager,
+            connection_manager,
+        )
+        .map_err(CommsBuilderError::OutboundMessagePoolError)
+    }
+
+    pub fn make_inbound_message_service(
+        &mut self,
+        node_identity: Arc<NodeIdentity<CommsPublicKey>>,
+        message_sink_address: InprocAddress,
+        dispatcher: DomainMessageDispatcher<MType>,
+        oms: OutboundMessageService,
+        peer_manager: Arc<PeerManager<CommsPublicKey, CommsDataStore>>,
+    ) -> Result<InboundMessageService<MType>, CommsBuilderError>
+    {
+        InboundMessageService::new(
+            self.comms_context.clone(),
+            node_identity,
+            message_sink_address,
+            Arc::new(construct_comms_msg_dispatcher()),
+            Arc::new(dispatcher),
+            Arc::new(oms),
+            peer_manager,
+        )
+        .map_err(CommsBuilderError::InboundMessageServiceError)
+    }
+
+    pub fn build(mut self) -> Result<CommsServices<MType>, CommsBuilderError> {
+        let node_identity = self.make_node_identity()?;
+
+        let peer_manager = self.make_peer_manager()?;
+
+        let peer_conn_config = self.make_peer_connection_config();
+
+        let control_service = self.make_control_service(node_identity.clone());
+
+        let connection_manager =
+            self.make_connection_manager(node_identity.clone(), peer_manager.clone(), peer_conn_config.clone());
+
+        let outbound_message_sink_address = InprocAddress::random();
+        let oms = self.make_oms(
+            node_identity.clone(),
+            outbound_message_sink_address.clone(),
+            peer_manager.clone(),
+        )?;
+
+        let omp = self.make_omp(
+            outbound_message_sink_address,
+            peer_manager.clone(),
+            connection_manager.clone(),
+        )?;
+
+        let dispatcher = self.dispatcher_factory.make();
+
+        let ims = self.make_inbound_message_service(
+            node_identity,
+            peer_conn_config.message_sink_address,
+            dispatcher,
+            oms,
+            peer_manager.clone(),
+        )?;
+
+        Ok(CommsServices {
+            control_service,
+            ims,
+            connection_manager,
+            omp,
+        })
+    }
+}
+
+#[derive(Debug, Error)]
+pub enum CommsServicesError {
+    ControlServiceError(ControlServiceError),
+    ConnectionManagerError(ConnectionError),
+    /// Comms services shut down uncleanly
+    UncleanShutdown,
+}
+
+pub struct CommsServices<MType>
+where
+    MType: Serialize + DeserializeOwned,
+    MType: DispatchableKey,
+    MType: Clone,
+{
+    connection_manager: Arc<ConnectionManager>,
```

Comment:
I think you are missing the OMS in this collection? It doesn't get started like the others but it is integral to using the comms layer.

Reply:
Not needed in the current state of the code, but agree will be needed later - will leave it for another PR once we're there with the IMS dispatching changes